tests/Unit/MicroplateTest.php

```diff
@@ -0,0 +1,324 @@
+    {
+        $microplateCoordinate = new Coordinate('H', 2);
+
+        // test that this coordinate can be added to a 96 microplate
+        $microplate96Well = new Microplate(new CoordinateSystem96Well());
+        $microplate96Well->addWell($microplateCoordinate, 'foo');
+
+        // test that this coordinate can not be added to a 12 microplate
+        $this->expectExceptionObject(new \InvalidArgumentException('Expected a coordinate between as A1 and C4 for Mll\Microplate\CoordinateSystem12Well, got: H2.'));
+        $microplate12Well = new Microplate(new CoordinateSystem12Well());
+        $microplate12Well->addWell($microplateCoordinate, 'foo');
+    }
+
+    public function testThrowOnAddingToTheSameCoordinate(): void
+    {
+        $coordinateSystem = new CoordinateSystem96Well();
+
+        $microplate = new Microplate($coordinateSystem);
+
+        $microplateCoordinate1 = new Coordinate('A', 2);
+        $microplateCoordinate2 = new Coordinate('A', 2);
+
+        $microplate->addWell($microplateCoordinate1, 'foo');
+
+        $this->expectException(WellNotEmptyException::class);
+        $microplate->addWell($microplateCoordinate2, 'bar');
+    }
+
+    public function testSortWells(): void
+    {
+        $microplate = $this->preparePlate();
+
+        $keysSortedByRow = $microplate->sortedWells(FlowDirection::ROW())->keys();
+        self::assertSame('A1', $keysSortedByRow[0]);
+        self::assertSame('A2', $keysSortedByRow[1]);
+        self::assertSame('A3', $keysSortedByRow[2]);
+        self::assertSame('A4', $keysSortedByRow[3]);
+        self::assertSame('H11', $keysSortedByRow[94]);
+        self::assertSame('H12', $keysSortedByRow[95]);
+
+        $keysSortedByColumn = $microplate->sortedWells(FlowDirection::COLUMN())->keys();
+        self::assertSame('A1', $keysSortedByColumn[0]);
+        self::assertSame('B1', $keysSortedByColumn[1]);
+        self::assertSame('C1', $keysSortedByColumn[2]);
+        self::assertSame('D1', $keysSortedByColumn[3]);
+        self::assertSame('G12', $keysSortedByColumn[94]);
+        self::assertSame('H12', $keysSortedByColumn[95]);
+    }
+
+    public function testFreeWells(): void
+    {
+        $microplate = $this->preparePlate();
+
+        self::assertTrue(
+            $microplate->wells->some(
+                /**
+                 * @param mixed|null $value
+                 * @param string $key
+                 *
+                 * @return bool
+                 */
+                fn ($value, $key) => null !== $value
+            )
+        );
+        self::assertNotCount(0, $microplate->freeWells());
+        $microplate->freeWells()->map(
+            /**
+             * @param null $value
+             * @param string $key
+             *
+             * @return void
+             */
+            fn ($value, $key) => self::assertNull($value)
+        );
+    }
+
+    /**
+     * @phpstan-return Microplate<mixed>
```

Comment:
```suggestion
     * @return Microplate<int|null>
```

Also, shouldn't Microplate be parameterized by the type of the coordinate system?